setup.py

```diff
@@ -37,14 +37,14 @@
     # requirements
     python_requires='>=3.6.*',
     install_requires=[
-        'requests>=2.22.0',
-        'keyring==12.0.2',
-        'deprecated==1.2.4',
+        'requests>=2.22.0,<3.0',
+        'keyring>=12.0.2',
```

Comment:
Some later versions of keying had breaking changes, which is why we required a specific version we knew worked. I would recommend verifying that the latest version works as expected on Windows, Linux (We used Ubuntu distro for testing previously), and Mac before requiring the latest version.

Specifically on Linux, there's no default credential storage, unlike Mac and Windows, so we would also need to verify that it can work (defaulting to a text file storage) on ones without a keychain (I believe Ubuntu comes with GNOME keyring). This might occur on something like a EC2 AmazonLinux instance or a minimal linux install. 

We've also had issues with headless connections (e.g. SSH) on some version of keyring version >12 when a cred storage was available on the Linux machine, but the user was unable to unlock the cred store w/out a pop-up GUI password prompt. I'm unsure if this was addressed in the latest version of keyring.

Reply:
I've re-pinned keyring given that it may have issues beyond what is covered in unit and integration tests and created an issue to investigate upgrading it in the future. https://sagebionetworks.jira.com/browse/SYNPY-1077

keyring 12.0.2 has no downstream dependencies that aren't already maxxed out (SecretStorage < 3 on Linux) so at least its downstream dependencies shouldn't shift underneath it if it is pinned.